docs/sts/client-grants.md

```diff
@@ -0,0 +1,83 @@
+
+Calling AssumeRoleWithClientGrants does not require the use of Minio default credentials. Therefore, you can distribute an application that requests temporary security credentials without including Minio default credentials in the application. Instead, the identity of the caller is validated by using a JWT access token from the identity provider. The temporary security credentials returned by this API consist of an access key, a secret key, and a security token. Applications can use these temporary security credentials to sign calls to Minio API operations.
+
+By default, the temporary security credentials created by AssumeRoleWithClientGrants last for one hour. However, you can use the optional DurationSeconds parameter to specify the duration of your session. You can provide a value from 900 seconds (15 minutes) up to the maximum session duration to 12 hours.
+
+### Request Parameters
+#### DurationSeconds
+The duration, in seconds. The value can range from 900 seconds (15 minutes) up to the 12 hours. If you specify a value higher than this setting, the operation fails. By default, the value is set to 3600 seconds.
+
+| Params | Value |
+| :-- | :-- |
+| *Type* | *Integer* |
+| *Valid Range* | *Minimum value of 900. Maximum value of 43200.* |
+| *Required* | *No* |
+
+#### Token
+The OAuth 2.0 access token that is provided by the identity provider. Your application must get this token by authenticating the application using client grants before the application makes an AssumeRoleWithClientGrants call.
+
+| Params | Value |
+| :-- | :-- |
+| *Type* | *String* |
+| *Length Constraints* | *Minimum length of 4. Maximum length of 2048.* |
+| *Required* | *Yes* |
+
+#### Response Elements
+XML response for this API is similar to [AWS STS AssumeRoleWithWebIdentity](https://docs.aws.amazon.com/STS/latest/APIReference/API_AssumeRoleWithWebIdentity.html#API_AssumeRoleWithWebIdentity_ResponseElements)
+
+#### Errors
+XML error response for this API is similar to [AWS STS AssumeRoleWithWebIdentity](https://docs.aws.amazon.com/STS/latest/APIReference/API_AssumeRoleWithWebIdentity.html#API_AssumeRoleWithWebIdentity_Errors)
+
+#### Testing
+Start Minio server with etcd configured, please follow installation guide here [etcd](https://coreos.com/etcd/docs/latest/). Minio STS feature requires etcdv3 server running.
+```
+$ export MINIO_ACCESS_KEY=minio
+$ export MINIO_SECRET_KEY=minio123
+$ export MINIO_ETCD_ENDPOINTS=http://127.0.0.1:2379
+$ export MINIO_IAM_JWKS_URL=https://localhost:9443/oauth2/jwks
+$ export MINIO_IAM_OPA_URL=http://localhost:8181/v1/data/httpapi/authz
+$ minio server /mnt/export
+
+$ ETCDCTL_API=3 etcdctl get /config/iam/iam.json --print-value-only | jq .
+{
+  "version": "1",
+  "identity": {
+    "type": "openid",
+    "openid": {
+      "jwt": {
+        "webKeyURL": "https://localhost:9443/oauth2/jwks"
```

Comment:
Any reason you changed it from your previous suggestion:

```
            "jwks": {
                "url": "http://192.168.1.12:9763/oauth2/jwks"
            }
```

https://auth0.com/docs/jwks

"url" or "endpoint" is better than unfamiliar and more complicated `webKeyURL`

`jwt` indicates that it's a token, hence `jwtConfig` or `jwks` is better

Reply:
```
                  OpenID struct {                                                                                                                                                                                
                          JWT validator.JWTArgs `json:"jwt"`                                                                                                                                                     
                  } `json:"openid"`                                                                                                                                                                              
```
This is also not consistent with `MINIO_IAM_JWKS_URL`

Reply:
Will add JWKS instead.